src/target/kinetis.c

```diff
@@ -152,6 +163,28 @@ bool kinetis_probe(target *t)
 		kl_gen_add_flash(t, 0, 0x40000, 0x800);
 		kl_gen_add_flash(t, 0x40000, 0x40000, 0x800);
 		break;
+	case 0x0: /* Chips that do not support [FAMILYID SUBFAMID SERIESID] fields. */
+
+		/* MK20 has 0x1 in FAMID (bits 4-6) */
+		if(((sdid >> 4) & 0x7) == 0x1){
+			t->driver = "MK20";
+			const uint32_t sram_l_start = 0x1C000000;
```

Comment:
Why not just put the base and length as call parameters.  They're no less magic this way.
The length here is also one too low.